gateway/api_definition.go

```diff
@@ -1728,13 +1728,24 @@ func (a *APISpec) getVersionFromRequest(r *http.Request) string {
 		// First non-empty part of the path is the version ID
 		for _, part := range strings.Split(uPath, "/") {
 			if part != "" {
-				if a.VersionDefinition.StripVersioningData || a.VersionDefinition.StripPath {
+				matchesUrlVersioningPattern := true
+				if a.VersionDefinition.UrlVersioningPattern != "" {
+					re, err := regexp.Compile(a.VersionDefinition.UrlVersioningPattern)
```

Comment:
@andrei-tyk how about we compile the regexp and keep it as a dependency on `APISpec`, so that it doesn't get compiled on every request affecting latency?

Reply:
the regexp package handles caching so you don't need to, just make sure it's used